lm_eval/models/fragma.py

```diff
@@ -42,18 +42,30 @@ def generate_until(self, reqs: list[Instance]) -> list[str]:
                     }
                 ]
             }
-
-            response = requests.post(self.endpoint,
-                                     headers=self.headers,
-                                     json=data)
-
-            if response.status_code != 200:
-                print(f"An error occurred while generating: {response.text}")
-                raise Exception(response.text)
-
-            result = response.json()
-            results.append(result['choices'][0]['message']["content"])
-
+            max_attempts = 10
+            attempts = 0
+            while attempts < max_attempts:
+                try:
+                    response = requests.post(self.endpoint,
+                                             headers=self.headers,
+                                             json=data)
+
+                    if response.status_code != 200:
+                        print(f"An error occurred while generating: {response.text}")
+                        raise Exception(response.text)
+
+                    result = response.json()
+                    results.append(result['choices'][0]['message']["content"])
+                    attempts = 0
+                    break
+                except Exception as e:
+                    attempts += 1
+                    print("error occured. retrying...")
+                    continue
+
+            if attempts == max_attempts:
+                print("maximum retries occured. Insert dummy result...")
+                results.append("dummy text")
```

Comment:
May I know what is the reason for adding `"dummy text"`?

Reply:
If we raise error when it occurs maximum retries, then previous inferences will be thrown. to avoid this, I try to pass the error sample with "dummy text" and I plan to filter out those sample later.